lib/ansible/module_utils/network/tetration.py

```diff
@@ -0,0 +1,381 @@
+            if 'offset' in query_result and not search_array:
+                params['offset'] = query_result['offset']
+            else:
+                return result_array if result_array else None
+
+    def run_method(self, method_name, target, **kwargs):
+        methods = {
+            'get': self.get,
+            'post': self.post,
+            'put': self.put,
+            'delete': self.delete
+        }
+        if 'req_payload' in kwargs:
+            kwargs['json_body'] = json.dumps(kwargs['req_payload'])
+            del kwargs['req_payload']
+        return methods[method_name.lower()](target, **kwargs)
+
+    def get(self, target, **kwargs):
+        resp = self.rc.get(target, **kwargs)
+        # import pdb; pdb.set_trace()
+        if resp.status_code == 400:
+            return None
+        elif resp.status_code == 200:
+            return resp.json()
+        else:
+            self.handle_exception('get', resp)
+
+    def post(self, target, **kwargs):
+        resp = self.rc.post(target, **kwargs)
+        if resp.status_code / 100 == 2:
+            try:
+                return resp.json()
+            except ValueError:
+                return None
+        else:
+            self.handle_exception('post', resp)
+
+    def put(self, target, **kwargs):
+        resp = self.rc.put(target, **kwargs)
+        if resp.status_code / 100 == 2:
+            try:
+                return resp.json()
+            except ValueError:
+                return None
+        else:
+            self.handle_exception('put', resp)
+
+    def delete(self, target, **kwargs):
+        resp = self.rc.delete(target, **kwargs)
+        if resp.status_code / 100 == 2:
+            try:
+                return resp.json()
+            except ValueError:
+                return None
+        else:
+            self.handle_exception('delete', resp)
+
+    def filter_object(self, obj1, obj2, check_only=False):
+        changed_flag = False
+        try:
+            for k in list(iterkeys(obj1)):
+                if k in list(iterkeys(obj2)):
+                    if type(obj1[k]) is dict:
+                        if (lambda a, b: (a > b) - (a < b))(obj1[k], obj2[k]) != 0:
+                            changed_flag = True
+                    elif obj1[k] != obj2[k]:
+                        changed_flag = True
+                    if not changed_flag and not check_only:
+                        del obj1[k]
+                else:
+                    changed_flag = True
+            return changed_flag
+        except AttributeError:
+            changed_flag = True
+            return changed_flag
+
+    def compare_keys(self, obj1, obj2):
+        unknown_keys = []
+        for k in list(iterkeys(obj1)):
+            if k not in list(iterkeys(obj2)):
+                unknown_keys.append(k)
+        return unknown_keys
+
+    def clear_values(self, obj):
+        for k in list(iterkeys(obj)):
+            obj[k] = ''
+
+
+TETRATION_API_PROTOCOLS = [
```

Comment:
Is this used somewhere that I'm not seeing?

Reply:
The `TETRATION_API_PROTOCOLS` variable
is used in other modules we plan to submit once this initial submission is
complete.